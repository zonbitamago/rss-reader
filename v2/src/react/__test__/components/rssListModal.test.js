'use strict';
jest.dontMock('../../components/rssListModal.jsx');
import React from 'react';
import renderer from 'react-test-renderer';
import $ from 'jquery';
import {mount} from 'enzyme';
import RssListModal from '../../components/rssListModal.jsx';

const open = false;
const onRssListModalClick = jest.fn();
const onRssListDeleteClick = jest.fn();
const actions = {
  onRssListModalClick: onRssListModalClick,
  onRssListDeleteClick: onRssListDeleteClick
};

beforeEach(() => {
  document.body.innerHTML = ''
})

test('snapshot', function() {
  const tree = renderer.create(<RssListModal actions={actions} open={false}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('modal open', () => {
  const wrapper = mount(<RssListModal actions={actions} open={true}/>);
  var modal = document.querySelector('.ui.page.modals.dimmer.transition.visible.active').innerHTML
  expect(modal).toMatchSnapshot();
})

test('modal close button', () => {
  const wrapper = mount(<RssListModal actions={actions} open={true}/>);
  var modal = document.querySelector('.ui.page.modals.dimmer.transition.visible.active')
  modal.querySelector('.ui.red.button').click();
  expect(onRssListModalClick).toHaveBeenCalled();
})

describe('rssList', () => {
  describe('表示', () => {

    test('rssListが空の場合は何も表示されない', () => {
      var rssList = undefined;
      const wrapper = mount(<RssListModal actions={actions} open={true} rssList={rssList}/>);
      var rssList = document.querySelectorAll('body > div.ui.page.modals.dimmer.transition.visible.active > div > div.scrolling.content > div > ul:nth-child(1) > li');
      expect(rssList.length).toEqual(0);
    });

    test('rssListが1つの場合は1つ表示される', () => {
      var rssList = [
        {
          name: 'name1',
          url: 'url1'
        }
      ];
      const wrapper = mount(<RssListModal actions={actions} open={true} rssList={rssList}/>);
      var rssList = document.querySelectorAll('body > div.ui.page.modals.dimmer.transition.visible.active > div > div.scrolling.content > div > ul:nth-child(1) > li');
      expect(rssList.length).toEqual(1);
    });

    test('rssListが2つの場合は2つ表示される', () => {
      var rssList = [
        {
          name: 'name1',
          url: 'url1'
        }, {
          name: 'name2',
          url: 'url2'
        }
      ];
      const wrapper = mount(<RssListModal actions={actions} open={true} rssList={rssList}/>);
      var rssList = document.querySelectorAll('body > div.ui.page.modals.dimmer.transition.visible.active > div > div.scrolling.content > div > ul:nth-child(1) > li');
      expect(rssList.length).toEqual(2);
    });

  });

  describe('削除', () => {
    test('削除処理が呼び出される。', () => {
      var rssList = [
        {
          name: 'name1',
          url: 'url1'
        }
      ];
      localStorage.setItem('rssList', rssList);
      const wrapper = mount(<RssListModal actions={actions} open={true} rssList={rssList}/>);
      var deleteButton = document.querySelector('body > div.ui.page.modals.dimmer.transition.visible.active > div > div.scrolling.content > div > ul:nth-child(1) > li:nth-child(1) > button');
      deleteButton.click();
      expect(onRssListDeleteClick).toHaveBeenCalled();

    })
  });

})
