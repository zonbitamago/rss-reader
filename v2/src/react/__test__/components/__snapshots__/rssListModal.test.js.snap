// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal open 1`] = `"<div class=\\"ui modal transition visible active background\\"><div class=\\"header\\">RssList</div><div class=\\"scrolling content\\"><div><ul class=\\"rssInputList\\"><li><button class=\\"ui blue mini button\\" role=\\"button\\"><i aria-hidden=\\"true\\" class=\\"remove icon\\"></i>Delete</button><a href=\\"#\\">test</a></li></ul><ul class=\\"rssInputList\\"><li><div class=\\"ui left labeled input rssInputTag\\"><div class=\\"ui tag label\\">Name</div><input type=\\"text\\" value=\\"\\" placeholder=\\"Enter Name\\"></div></li><li><div class=\\"ui left labeled input rssInputTag\\"><div class=\\"ui tag label\\">URL</div><input type=\\"text\\" value=\\"\\" placeholder=\\"Enter URL\\"></div></li></ul></div></div><div class=\\"actions\\"><button class=\\"ui red button\\" role=\\"button\\"><i aria-hidden=\\"true\\" class=\\"remove icon\\"></i>No</button><button class=\\"ui green button\\" role=\\"button\\"><i aria-hidden=\\"true\\" class=\\"checkmark icon\\"></i>Yes</button></div></div>"`;

exports[`snapshot 1`] = `
<i
  aria-hidden="true"
  className="yellow rss icon"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
/>
`;
